File: app/src/main/java/dev/ragnarok/fenrir/mvp/view/IMessageAttachmentsView.java
package dev.ragnarok.fenrir.mvp.view;

import android.net.Uri;

import androidx.annotation.NonNull;

import java.util.List;

import dev.ragnarok.fenrir.model.AttachmentEntry;
import dev.ragnarok.fenrir.model.LocalPhoto;
import dev.ragnarok.fenrir.model.ModelsBundle;
import dev.ragnarok.fenrir.mvp.core.IMvpView;


public interface IMessageAttachmentsView extends IMvpView, IErrorView {

    void displayAttachments(List<AttachmentEntry> entries);

    void notifyDataAdded(int positionStart, int count);

    void addPhoto(int accountId, int ownerId);

    void notifyEntryRemoved(int index);

    void displaySelectUploadPhotoSizeDialog(List<LocalPhoto> photos);

    void displayCropPhotoDialog(Uri uri);

    void displaySelectUploadFileSizeDialog(String file);

    void changePercentageSmoothly(int id, int progress);

    void notifyItemChanged(int index);

    void setEmptyViewVisible(boolean visible);

    void requestCameraPermission();

    void startCamera(@NonNull Uri fileUri);

    void syncAccompanyingWithParent(ModelsBundle accompanying);

    void startAddDocumentActivity(int accountId);

    void startAddVideoActivity(int accountId, int ownerId);

    void startAddAudioActivity(int accountId);
}

File: app/src/main/java/dev/ragnarok/fenrir/mvp/view/IMessageAttachmentsView.kt
package dev.ragnarok.fenrir.mvp.view

import android.net.Uri
import dev.ragnarok.fenrir.model.AttachmentEntry
import dev.ragnarok.fenrir.model.LocalPhoto
import dev.ragnarok.fenrir.model.ModelsBundle
import dev.ragnarok.fenrir.mvp.core.IMvpView

interface IMessageAttachmentsView : IMvpView, IErrorView {
    fun displayAttachments(entries: List<AttachmentEntry>)
    fun notifyDataAdded(positionStart: Int, count: Int)
    fun addPhoto(accountId: Int, ownerId: Int)
    fun notifyEntryRemoved(index: Int)
    fun displaySelectUploadPhotoSizeDialog(photos: List<LocalPhoto>)
    fun displayCropPhotoDialog(uri: Uri)
    fun displaySelectUploadFileSizeDialog(file: String)
    fun changePercentageSmoothly(id: Int, progress: Int)
    fun notifyItemChanged(index: Int)
    fun setEmptyViewVisible(visible: Boolean)
    fun requestCameraPermission()
    fun startCamera(fileUri: Uri)
    fun syncAccompanyingWithParent(accompanying: ModelsBundle)
    fun startAddDocumentActivity(accountId: Int)
    fun startAddVideoActivity(accountId: Int, ownerId: Int)
    fun startAddAudioActivity(accountId: Int)
}